src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandlerWithIncremental.cs

```diff
@@ -88,6 +92,28 @@ private void ReloadModelsPerChanges(IEnumerable<HostService> hostServices)
             }
         }
 
+        private void RegisterUnloadedXRefSpec()
+        {
+            var lastXrefMap = LastBuildVersionInfo?.XRefSpecMap;
+            foreach (var m in _unloadedFiles)
+            {
+                if (lastXrefMap == null)
+                {
+                    throw new BuildCacheException($"Full build hasn't loaded XRefMap.");
+                }
```

Comment:
only check this when there is any hostservice that can incremental

---
In reply to: [90367913](https://github.com/dotnet/docfx/pull/936#discussion_r90367913) [](ancestors = 90367913)